lib/async.js

```diff
@@ -781,7 +781,7 @@
                 } else if (test.apply(this, args)) {
                     iterator(next);
                 } else {
-                    callback(null);
+                    callback(null, args[0]);
```

Comment:
Is there a reason to only pass on 1 argument instead of all the arguments that were passed to the `cb`?

For example `waterfall` passes on all the arguments from `cb` on to the next function.